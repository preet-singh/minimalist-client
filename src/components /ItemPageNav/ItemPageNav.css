.ItemPageNav__back-button {
    margin: 45px 20px 25px auto
  }

.ItemPageNav__back-button:hover {
    background-color: #FFB72F;
  }
  
.ItemPageNav__inventory-name {
    color: #b5b3b3;
    font-family: var(--custom-font);
    font-size: 40px;
    height: 50vh;
    margin: 0 auto 0 auto;
    text-orientation: upright;
    width: 75px;
    writing-mode: vertical-lr;
  }
  