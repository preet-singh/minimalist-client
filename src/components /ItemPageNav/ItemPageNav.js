import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Button from '../Button/Button'
import InventoryContext from '../../InventoryContext'
import { findItem, findInventory } from '../item-helpers'
import PropTypes from 'prop-types';
import './ItemPageNav.css'


export default class ItemPageNav extends React.Component {
  static defaultProps = {
    history: {
      goBack: () => { }
    },
    match: {
      params: {}
    }
  }
  static contextType = InventoryContext;

  render() {
    const { items, inventory, } = this.context
    const { itemId } = this.props.match.params
    const item = findItem(items, itemId) || {}
    const inven = findInventory(inventory, item.inventory_id)
    return (
      <div className='ItemPageNav'>
        <Button
          tag='button'
          role='link'
          onClick={() => this.props.history.goBack()}
          className='ItemPageNav__back-button'
        >
          <FontAwesomeIcon icon='chevron-left' />
          <br />
          Back
        </Button>
        {inventory && (
          <h3 className='ItemPageNav__inventory-name'>
            {inventory.inventory_name}
          </h3>
        )}
      </div>
    )
  }
}

ItemPageNav.propTypes = {
  history: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired
}