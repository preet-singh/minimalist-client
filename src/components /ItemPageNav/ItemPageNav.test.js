import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ItemPageNav from './ItemPageNav'

describe(`ItemPageNav component`, () => {
  it('renders a .ItemPageNav by default', () => {
    const wrapper = shallow(<ItemPageNav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a h3 with inventory name when in props', () => {
    const props = {
      match: {
        params: {
          ItemId: 'test-item-id'
        }
      }
    }
    const context = {
      Items: [{ id: 'test-item-id', folderId: 'test-inventory-id' }],
      inventory: [{ id: 'test-inventory-id', name: 'Important' }]
    }

    const h3 = shallow(<ItemPageNav {...props} />, context)
      .find('.ItemPageNav__inventory-name')
    expect(toJson(h3)).toMatchSnapshot()
  })
})
