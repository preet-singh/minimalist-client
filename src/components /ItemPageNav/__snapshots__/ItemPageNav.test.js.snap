// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPageNav component renders a .ItemPageNav by default 1`] = `
<div
  className="ItemPageNav"
>
  <NavButton
    className="ItemPageNav__back-button"
    onClick={[Function]}
    role="link"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="chevron-left"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <br />
    Back
  </NavButton>
</div>
`;

exports[`ItemPageNav component renders a h3 with inventory name when in props 1`] = `null`;
